### PR TITLE
Ensure AKO reboot on secret Add/Delete

### PR DESCRIPTION
**Pull request by chauhanshubham**

### CODE REVIEW

File: internal/k8s/controller.go [hunk 1]
```diff
@@ -1034,6 +1034,9 @@ func validateAviConfigMap(obj interface{}) (*corev1.ConfigMap, bool) {
 
 func validateAviSecret(secret *corev1.Secret) bool {
 	if secret.Namespace == utils.GetAKONamespace() && secret.Name == lib.AviSecret {
+		// if the secret is updated or deleted we shutdown API server
```
Comment (monotosh-avi) [inline internal/k8s/controller.go:1037]: Though currently we are calling validateAviSecret() when secret gets added/updated, one can potentially call function from anywhere, is this the right place to trigger a shutdown ?

Comment (chauhanshubham) [reply internal/k8s/controller.go:1037]: This is on delete/update secret, specifically for `avi-secret`. do you see any issues around calling this from the event handler itself?

Comment (monotosh-avi) [reply internal/k8s/controller.go:1037]: Yeah, we can trigger a shutdown for the event handler.

**Review by monotosh-avi — APPROVED** [review]

**Review by hemantavi — APPROVED** [review]

**Comment by hemantavi** [comment]
build ako

**Pull request merged**